{
  "url": "https://api.github.com/repos/quarkusio/quarkus/issues/20357",
  "repository_url": "https://api.github.com/repos/quarkusio/quarkus",
  "labels_url": "https://api.github.com/repos/quarkusio/quarkus/issues/20357/labels{/name}",
  "comments_url": "https://api.github.com/repos/quarkusio/quarkus/issues/20357/comments",
  "events_url": "https://api.github.com/repos/quarkusio/quarkus/issues/20357/events",
  "html_url": "https://github.com/quarkusio/quarkus/issues/20357",
  "id": 1005361843,
  "node_id": "I_kwDOCFbutM477Jqz",
  "number": 20357,
  "title": "Resteasy Reactive: `ServerExceptionMapper` does not process a thrown `Exception` (Kotlin only)",
  "labels": [
    {
      "id": 985346620,
      "node_id": "MDU6TGFiZWw5ODUzNDY2MjA=",
      "url": "https://api.github.com/repos/quarkusio/quarkus/labels/kind/bug",
      "name": "kind/bug",
      "color": "d73a4a",
      "default": false,
      "description": "Something isn't working"
    },
    {
      "id": 1287515315,
      "node_id": "MDU6TGFiZWwxMjg3NTE1MzE1",
      "url": "https://api.github.com/repos/quarkusio/quarkus/labels/area/kotlin",
      "name": "area/kotlin",
      "color": "0366d6",
      "default": false,
      "description": ""
    },
    {
      "id": 2552031458,
      "node_id": "MDU6TGFiZWwyNTUyMDMxNDU4",
      "url": "https://api.github.com/repos/quarkusio/quarkus/labels/area/rest",
      "name": "area/rest",
      "color": "0366d6",
      "default": false,
      "description": ""
    }
  ],
  "state": "closed",
  "locked": false,
  "milestone": {
    "url": "https://api.github.com/repos/quarkusio/quarkus/milestones/151",
    "html_url": "https://github.com/quarkusio/quarkus/milestone/151",
    "labels_url": "https://api.github.com/repos/quarkusio/quarkus/milestones/151/labels",
    "id": 7197992,
    "node_id": "MI_kwDOCFbutM4AbdUo",
    "number": 151,
    "title": "2.3.0.Final",
    "description": "",
    "creator": {
      "login": "geoand",
      "id": 4374975,
      "node_id": "MDQ6VXNlcjQzNzQ5NzU=",
      "avatar_url": "https://avatars.githubusercontent.com/u/4374975?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/geoand",
      "html_url": "https://github.com/geoand",
      "followers_url": "https://api.github.com/users/geoand/followers",
      "following_url": "https://api.github.com/users/geoand/following{/other_user}",
      "gists_url": "https://api.github.com/users/geoand/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/geoand/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/geoand/subscriptions",
      "organizations_url": "https://api.github.com/users/geoand/orgs",
      "repos_url": "https://api.github.com/users/geoand/repos",
      "events_url": "https://api.github.com/users/geoand/events{/privacy}",
      "received_events_url": "https://api.github.com/users/geoand/received_events",
      "type": "User",
      "site_admin": false
    },
    "open_issues": 0,
    "closed_issues": 55,
    "state": "closed",
    "created_at": "2021-09-28T15:55:23Z",
    "updated_at": "2021-11-30T14:44:31Z",
    "due_on": null,
    "closed_at": "2021-10-06T08:40:32Z"
  },
  "comments": 4,
  "created_at": "2021-09-23T12:18:14Z",
  "updated_at": "2021-10-04T20:38:27Z",
  "closed_at": "2021-09-24T05:06:25Z",
  "active_lock_reason": null,
  "body": "### Describe the bug\n\nIf a (Kotlin-)Rest-Endpoint throws an `java.lang.Exception` then the `Exception` does not get processed by a custom `ServerExceptionMapper`. `RuntimeException` works as excepted.\r\n\r\n```kotlin\r\norg.acme.MyException\r\n\tat org.acme.ExceptionTestResource.throwException1Method(ExceptionTestResource.kt:33)\r\n\tat org.acme.ExceptionTestResource_Subclass.throwException1Method$$superforward1(ExceptionTestResource_Subclass.zig:113)\r\n\tat org.acme.ExceptionTestResource_Subclass$$function$$1.apply(ExceptionTestResource_Subclass$$function$$1.zig:24)\r\n\tat io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)\r\n\tat io.quarkus.arc.runtime.devconsole.InvocationInterceptor.proceed(InvocationInterceptor.java:62)\r\n\tat io.quarkus.arc.runtime.devconsole.InvocationInterceptor.monitor(InvocationInterceptor.java:49)\r\n\tat io.quarkus.arc.runtime.devconsole.InvocationInterceptor_Bean.intercept(InvocationInterceptor_Bean.zig:521)\r\n\tat io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)\r\n\tat io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)\r\n\tat io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)\r\n\tat org.acme.ExceptionTestResource_Subclass.throwException1Method(ExceptionTestResource_Subclass.zig:183)\r\n\tat org.acme.ExceptionTestResource_ClientProxy.throwException1Method(ExceptionTestResource_ClientProxy.zig:185)\r\n\tat org.acme.ExceptionTestResource$quarkusrestinvoker$throwException1Method_4a04d16e4f5785de322606c51c7fc425b79facd0.invoke(ExceptionTestResource$quarkusrestinvoker$throwException1Method_4a04d16e4f5785de322606c51c7fc425b79facd0.zig:33)\r\n\tat org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)\r\n\tat org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:7)\r\n\tat org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:141)\r\n\tat io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:543)\r\n\tat org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)\r\n\tat org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)\r\n\tat org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)\r\n\tat org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)\r\n\tat io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)\r\n\tat java.base/java.lang.Thread.run(Thread.java:829)\r\nResulted in: io.quarkus.arc.ArcUndeclaredThrowableException: Error invoking subclass method\r\n\tat org.acme.ExceptionTestResource_Subclass.throwException1Method(ExceptionTestResource_Subclass.zig:205)\r\n\t... 12 more\r\n```\n\n### Expected behavior\n\n_No response_\n\n### Actual behavior\n\n_No response_\n\n### How to Reproduce?\n\n1. https://code.quarkus.io/?e=resteasy-reactive-jackson&e=kotlin&extension-search=kotlin\r\n2. Add the following code\r\n\r\n```kotlin\r\npackage org.acme\r\n\r\nimport org.jboss.resteasy.reactive.server.ServerExceptionMapper\r\nimport javax.enterprise.context.ApplicationScoped\r\nimport javax.ws.rs.GET\r\nimport javax.ws.rs.Path\r\nimport javax.ws.rs.core.Response\r\n\r\nopen class SuperException : Exception() // Works if SuperException extends RuntimeException instead of Exception\r\nclass MyException : SuperException()\r\n\r\nclass ExceptionMappers {\r\n\r\n    @ServerExceptionMapper\r\n    fun superExceptionHandler(ex: SuperException): Response {\r\n        println(\"SuperException processed\")\r\n        return Response.serverError().build()\r\n    }\r\n}\r\n\r\n@ApplicationScoped\r\n@Path(\"/ex\")\r\nclass ExceptionTestResource {\r\n\r\n    /**\r\n     * Exception is not processed by the ServerExceptionMapper.\r\n     */\r\n    @GET\r\n    @Path(\"/throw1\")\r\n    fun throwException1Method(): String {\r\n        throw MyException()\r\n    }\r\n\r\n    /**\r\n     * Exception is processed by the ServerExceptionMapper.\r\n     */\r\n    @GET\r\n    @Path(\"/throw2\")\r\n    @Throws(MyException::class) // see here\r\n    fun throwExceptionMethod(): String {\r\n        throw MyException()\r\n    }\r\n}\r\n```\n\n### Output of `uname -a` or `ver`\n\n_No response_\n\n### Output of `java -version`\n\nJava 11\n\n### GraalVM version (if different from Java)\n\n_No response_\n\n### Quarkus version or git rev\n\nQuarkus 2.2.3.Final\n\n### Build tool (ie. output of `mvnw --version` or `gradlew --version`)\n\nMaven 3.8.1\n\n### Additional information\n\n_No response_",
  "reactions": {
    "url": "https://api.github.com/repos/quarkusio/quarkus/issues/20357/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/quarkusio/quarkus/issues/20357/timeline",
  "performed_via_github_app": null,
  "state_reason": "completed"
}
